src/client/app/components/RouteComponent.tsx

```diff
@@ -46,16 +46,19 @@ export default class RouteComponent extends React.Component<RouteProps, {}> {
 			redirectRoute();
 			return;
 		}
-		// Verify that the auth token is valid
-		axios.post('/api/verification/',
-			{ token: getToken() },
-			{ validateStatus: status => (status >= 200 && status < 300) || (status === 401 || status === 403) })
-			.then(res => {
-				// Route to login page if the auth token is not valid
-				if (!res.data.success) { browserHistory.push('/login'); }
-			})
-			// In the case of a server error, the user can't fix the issue. Log it for developers.
-			.catch(console.error); // eslint-disable-line no-console
+		// Verify that the auth token is valid.
+		// Needs to be async because of the network request
+		(async () => {
+			try {
+				if (!(await verificationApi.checkTokenValid())) {
+					// Route to login page if the auth token is not valid
+					browserHistory.push('/login');
+				}
+			} catch (e) {
+				// In the case of a server error, the user can't fix the issue. Log it for developers.
+				console.error(e); // tslint:disable-line no-console
```

Comment:
We have been using `throw new Error()` instead of logging directly to the console (as of the TypeScript PR).